src/meta/src/hummock/manager/mod.rs

```diff
@@ -353,7 +353,11 @@ impl HummockManager {
             if let risingwave_object_store::object::ObjectStoreImpl::S3(s3) = object_store.as_ref()
                 && !env.opts.do_not_config_object_storage_lifecycle
             {
-                s3.inner().configure_bucket_lifecycle().await;
+                let is_bucket_expiration_configured = s3.inner().configure_bucket_lifecycle().await;
+                if is_bucket_expiration_configured{
+                    return Err(ObjectError::internal("Bucket retention is already configured, the cluster is at risk of losing data")
```

Comment:
It is better to include the bucket name in the error message and ask user to disable retention manually. How about change the error message to
"Cluster cannot start with object expiration configured for bucket <xxx> because RisingWave data will be lost when object expiration kicks in. Please disable object expiration and restart the cluster."